### PR TITLE
Few managers optimizations

### PR DESCRIPTION
**Pull request by Abyss777**
Polished few things:

- Items cache:
  - Reorganized cache update, now it is similar to how we do it for Devices and Groups (do not clear items map, but update every item)
  - Used "overridable" functions `addNewItem`/`updateCachedItem` for two purposed update cache and update items. By default it just `put` but might be used to track additional maps (like uniqueIds) or update item by fields.
  - It is not necessary to override whole functions `addItem`/`updateItem`/`removeItem` and especially `refreshItems` just put manager specific code in `addNewItem`/`updateCachedItem`/`removeCachedItem` if necessary
  - Spread refresh calls by appropriate constructors

- Used `Permission` class also for retrieving permissions from database
  - Do not think we will use common function `executeMapQuery` in near future, replace it to particular `executePermissionsQuery`
  - Updating permission cache now looks much much clear.

Also fixed small issue in group cache update.

### CODE REVIEW

File: src/org/traccar/database/SimpleObjectManager.java [hunk 1]
```diff
@@ -65,11 +67,7 @@ protected final void clearItems() {
         items.clear();
     }
 
-    protected final void putItem(long itemId, BaseModel item) {
-        items.put(itemId, item);
-    }
-
-    protected final void removeCachedItem(long itemId) {
+    protected void removeCachedItem(long itemId) {
```
Comment (tananaev) [inline src/org/traccar/database/SimpleObjectManager.java]: I think it makes sense to keep `updateCachedItem` and `removeCachedItem` next to each other.

File: src/org/traccar/database/SimpleObjectManager.java [hunk 2]
```diff
@@ -91,9 +89,23 @@ public final boolean checkItemPermission(long userId, long itemId) {
...
+                    databaseItemIds.add(item.getId());
+                    if (items.containsKey(item.getId())) {
+                        updateCachedItem(item);
+                    } else {
+                        addNewItem(item);
+                    }
+                }
+                for (Long cachedItemId : items.keySet()) {
+                    if (!databaseItemIds.contains(cachedItemId)) {
+                        items.remove(cachedItemId);
```
Comment (tananaev) [inline src/org/traccar/database/SimpleObjectManager.java]: Should we use `removeCachedItem` here?

**Comment by Abyss777** [comment]
Thanks for the tip. Did not think about it. Now it is perfect.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**